src/decoding/extrinsic_type_info.rs

```diff
@@ -97,23 +108,32 @@ impl<'a> core::fmt::Display for ExtrinsicInfoError<'a> {
                 pallet_index,
                 pallet_name,
             } => {
-                write!(f, "Cannot find calls type with id {id} in pallet '{pallet_name}' (pallet index {pallet_index})")
+                write!(f, "Cannot find calls type with id {id} in pallet '{pallet_name}' (pallet index {pallet_index}).")
             }
             ExtrinsicInfoError::CallsTypeShouldBeVariant {
                 id,
                 pallet_index,
                 pallet_name,
             } => {
-                write!(f, "Calls type with id {id} should be a variant in pallet '{pallet_name}' (pallet index {pallet_index})")
+                write!(f, "Calls type with id {id} should be a variant in pallet '{pallet_name}' (pallet index {pallet_index}).")
             }
             ExtrinsicInfoError::ExtrinsicTypeNotFound { id } => {
-                write!(f, "Could not find the extrinsic type with id {id}")
+                write!(f, "Could not find the extrinsic type with id {id}.")
             }
             ExtrinsicInfoError::ExtrinsicAddressTypeNotFound => {
-                write!(f, "Could not find the extrinsic address type")
+                write!(f, "Could not find the extrinsic address type.")
             }
             ExtrinsicInfoError::ExtrinsicSignatureTypeNotFound => {
-                write!(f, "Could not find the extrinsic signature type")
+                write!(f, "Could not find the extrinsic signature type.")
+            }
+            ExtrinsicInfoError::ExtrinsicExtensionVersionNotSupported { extension_version } => {
+                // Dev note: If we see a V5 GEneral extrinsic, it will have a byte for the version of the transaction extensions.
```

Comment:
You could clarify in the comment that is possible that `metadata < v16` can have the transaction extension version == 0 which we know how to decode but `metadata >= v16` i.e, "V5 General extrinsic" we don't know how to decode yet...

However, you added a similar comment in `err_if_bad_extension_version` which was easier for me to understand.

Reply:
I had a go at tweaking the wording, but yeah it's a bit tricky to word correctly :D